addons/website/models/website.py

```diff
@@ -437,6 +462,199 @@ def action_dashboard_redirect(self):
 
 
+class SeoMetadata(models.AbstractModel):
+
+    _name = 'website.seo.metadata'
+    _description = 'SEO metadata'
+
+    website_meta_title = fields.Char("Website meta title", translate=True)
+    website_meta_description = fields.Text("Website meta description", translate=True)
+    website_meta_keywords = fields.Char("Website meta keywords", translate=True)
+
+
+class WebsitePublishedMixin(models.AbstractModel):
+
+    _name = "website.published.mixin"
+
+    website_published = fields.Boolean('Visible in Website', copy=False)
+    website_url = fields.Char('Website URL', compute='_compute_website_url', help='The full URL to access the document through the website.')
+
+    @api.multi
+    def _compute_website_url(self):
+        for record in self:
+            record.website_url = '#'
+
+    @api.multi
+    def website_publish_button(self):
+        self.ensure_one()
+        if self.env.user.has_group('website.group_website_publisher') and self.website_url != '#':
+            return self.open_website_url()
+        return self.write({'website_published': not self.website_published})
+
+    def open_website_url(self):
+        return {
+            'type': 'ir.actions.act_url',
+            'url': self.website_url,
+            'target': 'self',
+        }
+
+
+class Page(models.Model):
+    _name = 'website.page'
+    _inherits = {'ir.ui.view': 'ir_ui_view_id'}
+    _inherit = 'website.published.mixin'
+    _description = 'Page'
+
+    name = fields.Char('Page Name')
+    url = fields.Char('Page Url')
+    website_ids = fields.Many2many('website', string='Websites')
+    ir_ui_view_id = fields.Many2one('ir.ui.view', string='View', required=True, ondelete="cascade")
+    website_indexed = fields.Boolean('Page Indexed', default=True)
+    date_publish = fields.Datetime('Published Date')
+    # This is needed to be able to display if page is a menu in /website/page_management
+    menu_ids = fields.One2many('website.menu', 'page_id', 'Related Menus')
+    is_homepage = fields.Boolean(compute='_compute_homepage', string='Homepage')
+    is_visible = fields.Boolean(compute='_compute_visible', string='Is Visible')
+
+    @api.one
+    def _compute_homepage(self):
+        self.is_homepage = self == self.env['website'].get_current_website().homepage_id
+        
+    @api.one
+    def _compute_visible(self):
+        self.is_visible = self.website_published and (not self.date_publish or self.date_publish < fields.Datetime.now())
+
+    @api.model
+    def get_page_info(self, id, website_id):
+        domain = ['|', ('website_ids', 'in', website_id), ('website_ids', '=', False), ('id', '=', id)]
+        item = self.search_read(domain, fields=['id', 'name', 'url', 'website_published', 'website_indexed', 'date_publish', 'menu_ids', 'is_homepage'], limit=1)
+        return item
+
+    @api.model
+    def save_page_info(self, website_id, data):
+        website = self.env['website'].browse(website_id)
```

Comment:
Can we use write ?

Reply:
I don't think we can since it is getting some informations that are not directly website.page related like is_homepage & is_menu (which are non stored computed fields used to show that switch/toggle on page property dialog).
is_homepage at the end will write on website to set/unset website.homepage_id
is_menu at the end will delete website.menu having a m2o relation to that page or create a menu with a m2o to the page

Can I do all of that in an override of write ?